cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -728,18 +735,39 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
     private fun handleIf(node: Python.AST.If): Statement {
         val ret = newIfStatement(rawNode = node)
         ret.condition = frontend.expressionHandler.handle(node.test)
-        ret.thenStatement =
-            if (node.body.isNotEmpty()) {
+
+        // Since python is a dynamically invoked language, it would allow for different branches of
+        // the programs to declare different symbols. This is frequently used in the Python typeshed
+        // to declare different declarations based on the python version and the platform in a way
+        // that C/C++ would use preprocessor macro. Therefore, we replace occurrences of
+        // sys.platform and sys.version_info by their actual values when we encounter the
+        // expression.
+        //
+        // It would be too complex (=impossible) for _all_ if statements to be evaluated and decided
+        // which branch would be executed at runtime. But we can have a simple heuristic that
+        // evaluates simple constant expressions.
+        val constCondition = ret.condition?.evaluate(constEvaluator) as? Boolean
```

Comment:
I really dislike that we move such logic into the frontend. In my understanding, the frontends were meant to be simple and have a "stupid" translation of AST nodes to the CPG's AST version. If we now decide to remove some code, perform some preliminary evaluations and so on, this contradicts my understanding of the goal. I also thought that the CPG would keep all available information in the code and not remove parts of the code (in particular not in the frontend and without any way to disable this behavior).

Why don't we use a pass to declare this branch as unreachable and, based on this, remove the affected declarations (or also flag the refersTo/usages/... edges as inappropriate)? This would be more clean in my opinion and make it easier to understand the final result.

Reply:
100% agree. An optional pass executing early would be the preferred solution from my point of view, too. I do not like this in the frontend.

Reply:
I copy/pasted the old unreachable EOG pass for now, unfortunately this will not work on a large code base. Seems this is stuck until #1811

Reply:
In the meantime I will split some of the things in this PR into smaller ones.